src/main/java/org/fogbowcloud/manager/core/threads/FulfilledMonitor.java

```diff
@@ -0,0 +1,174 @@
+     * Attribute that represents the thread sleep time when there is no orders
+     * to be processed.
+     */
+    private Long sleepTime;
+
+    public FulfilledMonitor(InstanceProvider localInstanceProvider, InstanceProvider remoteInstanceProvider,
+                            TunnelingServiceUtil tunnelingService, SshConnectivityUtil sshConnectivity,
+                            Properties properties) {
+        this.localInstanceProvider = localInstanceProvider;
+        this.remoteInstanceProvider = remoteInstanceProvider;
+        this.localMemberId = properties.getProperty(ConfigurationConstants.XMPP_ID_KEY);
+
+        this.computeInstanceConnectivity = new ComputeInstanceConnectivityUtil(tunnelingService, sshConnectivity);
+
+        SharedOrderHolders sharedOrderHolders = SharedOrderHolders.getInstance();
+        this.fulfilledOrdersList = sharedOrderHolders.getFulfilledOrdersList();
+
+        String schedulerPeriodStr = properties.getProperty(ConfigurationConstants.FULFILLED_ORDERS_SLEEP_TIME_KEY,
+                DefaultConfigurationConstants.FULFILLED_ORDERS_SLEEP_TIME);
+        this.sleepTime = Long.valueOf(schedulerPeriodStr);
+    }
+
+    /**
+     * Iterates over the fulfilled orders list and try to process one fulfilled order per
+     * time. If the order is null it indicates the iteration is in the end of
+     * the list or the list is empty.
+     */
+    @Override
+    public void run() {
+        boolean isActive = true;
+
+        while (isActive) {
+            try {
+                Order order = this.fulfilledOrdersList.getNext();
+
+                if (order != null) {
+                    try {
+                        this.processFulfilledOrder(order);
+                    } catch (Throwable e) {
+                        LOGGER.error("Error while trying to process the order " + order, e);
+                    }
+                } else {
+                    this.fulfilledOrdersList.resetPointer();
+                    LOGGER.info("There is no fulfilled order to be processed, sleeping for "
+                            + this.sleepTime + " milliseconds");
+                    Thread.sleep(this.sleepTime);
+                }
+            } catch (InterruptedException e) {
+                isActive = false;
+                LOGGER.warn("Thread interrupted", e);
+            } catch (Throwable e) {
+                LOGGER.error("Not expected error", e);
+            }
+        }
+    }
+
+    /**
+     * Get an instance for a fulfilled order. If that instance is not reachable
+     * then order state is set to failed.
+     *
+     * @param order {@link Order}
+     * @throws OrderStateTransitionException Could not remove order from list of fulfilled orders.
+     */
+    protected void processFulfilledOrder(Order order) throws OrderStateTransitionException {
+        synchronized (order) {
+            OrderState orderState = order.getOrderState();
+
+            if (orderState.equals(OrderState.FULFILLED)) {
```

Comment:
You are not testing when the FulfilledMonitor try to process an order in a state that is not FULFILLED.

Reply:
Done!